sql/src/influxql/provider.rs

```diff
@@ -0,0 +1,466 @@
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn fields(&self) -> Vec<&ArrowField> {
+        self.columns
+            .iter()
+            .filter_map(|(influx_type, field)| {
+                if matches!(influx_type, InfluxColumnType::Field(..)) {
+                    Some(field)
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn time(&self) -> &ArrowField {
+        // Time column must exist, has checked it when building.
+        let time_column = self
+            .columns
+            .iter()
+            .find(|(influx_type, _)| matches!(influx_type, InfluxColumnType::Timestamp))
+            .unwrap();
+
+        &time_column.1
+    }
+
+    fn column(&self, idx: usize) -> (InfluxColumnType, &ArrowField) {
+        let (influx_type, field) = &self.columns[idx];
+
+        (*influx_type, field)
+    }
+
+    fn find_index_of(&self, name: &str) -> Option<usize> {
+        self.columns
+            .iter()
+            .enumerate()
+            .find(|(_, (_, field))| field.name() == name)
+            .map(|(index, _)| index)
+    }
+}
+
+fn map_column_type_to_influx_column_type(
+    column: &ColumnSchema,
+    is_timestamp_key: bool,
+) -> Result<InfluxColumnType> {
+    // Time column
+    if is_timestamp_key {
+        if column.name == "time" && !column.is_nullable {
+            Ok(InfluxColumnType::Timestamp)
+        } else {
+            BuildSchema {
+                msg: format!("invalid time column, column:{column:?}"),
+            }
+            .fail()
+        }
+    } else if column.is_tag {
+        // Tag column
+        if matches!(column.data_type, DatumKind::String) && column.is_nullable {
+            Ok(InfluxColumnType::Tag)
+        } else {
+            BuildSchema {
+                msg: format!("invalid tag column, column:{column:?}"),
+            }
+            .fail()
+        }
+    } else {
+        // Field column
+        map_field_type_to_influx_field_type(column).map(InfluxColumnType::Field)
+    }
+}
+
+fn map_field_type_to_influx_field_type(column: &ColumnSchema) -> Result<InfluxFieldType> {
+    if column.is_nullable {
+        match column.data_type {
+            DatumKind::Int64 => Ok(InfluxFieldType::Integer),
+            DatumKind::UInt64 => Ok(InfluxFieldType::UInteger),
+            DatumKind::Double => Ok(InfluxFieldType::Float),
+            DatumKind::String => Ok(InfluxFieldType::String),
+            DatumKind::Boolean => Ok(InfluxFieldType::Boolean),
+            DatumKind::Null
+            | DatumKind::Timestamp
+            | DatumKind::Float
+            | DatumKind::Varbinary
+            | DatumKind::UInt32
+            | DatumKind::UInt16
+            | DatumKind::UInt8
+            | DatumKind::Int32
+            | DatumKind::Int16
+            | DatumKind::Int8
+            | DatumKind::Date
+            | DatumKind::Time => BuildSchema {
```

Comment:
Why is conversion for these types disallowed?

Reply:
I add a comment to analyze this, because they are not supported in influxdb.
Maybe we can add a todo for supporting them?

Reply:
Ok. Let's add a todo here.